components/server/src/jobs/runner.ts

```diff
@@ -18,14 +18,15 @@ import { WebhookEventGarbageCollector } from "./webhook-gc";
 import { WorkspaceGarbageCollector } from "./workspace-gc";
 import { SnapshotsJob } from "./snapshots";
 import { OrgOnlyMigrationJob } from "./org-only-migration-job";
+import { JobStateDbImpl } from "@gitpod/gitpod-db/lib/typeorm/job-state-db-impl";
 
 export const Job = Symbol("Job");
 
-export interface Job {
+export interface Job<T = any> {
     readonly name: string;
     readonly frequencyMs: number;
     readonly lockedResources?: string[];
-    run: () => Promise<void>;
+    run: (previousState?: T) => Promise<T>;
```

Comment:
Do we need to push this into the Runner when we only have a single use-case for this today - the OrgOnlyMigrationJob?

I worry that for most cases, you shouldn't really rely on previous state and the migration should be setup in such a way that it can exclude the already migrated items and this almost makes it too easy to do the wrong thing

Reply:
I agree and will remove it with the migration job if we don't have any usage. We can always pull it back from history later.